src/java/main/org/apache/zookeeper/client/ConnectStringParser.java

```diff
@@ -68,14 +69,26 @@ public ConnectStringParser(String connectString) {
         List<String> hostsList = split(connectString,",");
         for (String host : hostsList) {
             int port = DEFAULT_PORT;
-            int pidx = host.lastIndexOf(':');
-            if (pidx >= 0) {
-                // otherwise : is at the end of the string, ignore
-                if (pidx < host.length() - 1) {
-                    port = Integer.parseInt(host.substring(pidx + 1));
-                }
-                host = host.substring(0, pidx);
+            if (!connectString.startsWith("[")) {//IPv4
```

Comment:
In https://github.com/apache/zookeeper/pull/548, it added ConfigUtils.getHostAndPort, which can split host port in IPV4 or IPV6 format, maybe we should reuse that code here.

Reply:
The client parse logic support the default port, so it's not easy to reuse the method ConfigUtils.getHostAndPort

Reply:
@lvfangmin  the client's parse logic supports default port, it's not easy to reuse the method getHostAndPort